coalib/results/SourceRange.py

```diff
@@ -120,6 +120,44 @@ def expand(self, file_contents):
                                        tr.end.line,
                                        tr.end.column)
 
+    @enforce_signature
+    def affected_source(self, file_dict: dict):
+        r"""
+        Tells which lines are affected in a specified file within a given range.
+
+        >>> from os.path import abspath
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=2)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n')})
```

Comment:
@sils don't we close the opening braces/parens in the same line as the last element? Or has that been relaxed?